papis/commands/update.py

```diff
@@ -86,100 +369,163 @@ def run(document: papis.document.Document,
-              nargs=2,
-              multiple=True,
-              default=(),)
-@click.option("-s", "--set", "set_tuples",
-              help="Update document's information with key value. "
-                   "The value can be a papis format.",
-              multiple=True,
-              type=(str, str),)
-@papis.cli.bool_flag("-b", "--batch",
-                     help="Batch mode, do not prompt or otherwise")
-def cli(query: str,
-        git: bool,
-        doc_folder: Tuple[str, ...],
-        from_importer: List[Tuple[str, str]],
-        batch: bool,
-        auto: bool,
-        _all: bool,
-        sort_field: Optional[str],
-        sort_reverse: bool,
-        set_tuples: List[Tuple[str, str]],) -> None:
-    """Update a document from a given library."""
-
-    documents = papis.cli.handle_doc_folder_query_all_sort(query,
-                                                           doc_folder,
-                                                           sort_field,
-                                                           sort_reverse,
-                                                           _all)
+@papis.cli.bool_flag("--auto", help="Try to parse information from different sources")
+@click.option(
+    "--from",
+    "from_importer",
+    help="Add document from a specific importer ({})".format(
+        ", ".join(papis.importer.available_importers())
+    ),
+    type=(click.Choice(papis.importer.available_importers()), str),
+    nargs=2,
+    multiple=True,
+    default=(),
+)
+@click.option(
+    "-s",
+    "--set",
+    "to_set",
+    help="Set a key's value. ",
+    multiple=True,
+    type=(str, str),
+)
+@click.option(
+    "-d",
+    "--drop",
+    "to_drop",
+    help="Drop a key-value pair.",
+    multiple=True,
+    type=str,
+)
+@click.option(
+    "-p",
+    "--append",
+    "to_append",
+    help="Append an item to a string or a list. ",
+    multiple=True,
+    type=(str, str),
+)
+@click.option(
+    "-r",
+    "--remove",
+    "to_remove",
+    help="Remove an item from a list.",
+    multiple=True,
+    type=(str, str),
+)
+@click.option(
+    "-n",
+    "--rename",
+    "to_rename",
+    help="Rename a list item.",
+    multiple=True,
+    type=(str, str, str),
+)
+@papis.cli.bool_flag("-b", "--batch", help="Batch mode, do not prompt or otherwise")
+def cli(
+    query: str,
+    git: bool,
+    doc_folder: Tuple[str, ...],
+    from_importer: List[Tuple[str, str]],
+    batch: bool,
+    auto: bool,
+    _all: bool,
+    sort_field: Optional[str],
+    sort_reverse: bool,
+    to_set: List[Tuple[str, str]],
+    to_drop: List[str],
+    to_append: List[Tuple[str, str]],
+    to_remove: List[Tuple[str, str]],
+    to_rename: List[Tuple[str, str, str]],
+) -> None:
+    """
+    Update a document from a given library. It accepts both Papis formatting
+    strings and Python expressions. Filenames are sanitised. See documentation
+    for more information.
```

Comment:
This seems rather long and not very consistent with the other commands?

Reply:
Ah, no no, I just meant the doc comment here. Most other commands have a one-line like
https://github.com/papis/papis/blob/4562963adedbf42ef70e88ebcc9b4fa8805c26f5/papis/commands/add.py#L501
because this will show up in `papis update --help`.